vm/src/vm/runners/cairo_runner.rs

```diff
@@ -185,7 +192,18 @@ impl CairoRunner {
             LayoutName::recursive_with_poseidon => CairoLayout::recursive_with_poseidon(),
             LayoutName::all_cairo => CairoLayout::all_cairo_instance(),
             LayoutName::all_solidity => CairoLayout::all_solidity_instance(),
-            LayoutName::dynamic => CairoLayout::dynamic_instance(),
+            LayoutName::dynamic => {
+                debug_assert!(
+                    cairo_layout_params.is_some(),
+                    "cairo layout params is missing while using dynamic layout"
+                );
+
+                let params = cairo_layout_params.ok_or(RunnerError::BadDynamicLayoutParams(
+                    "cairo layout param is missing".to_string(),
```

Comment:
This error string is used more than once, it could be its own error variant

Reply:
Done